src/main/java/edu/wisc/my/keyvalue/controller/KeyValueStoreController.java

```diff
@@ -57,6 +66,102 @@ public void setUsernameAttr(String attr) {
         }
     }
     
+    @RequestMapping(value="/{scope}/{key}", method=RequestMethod.PUT)
+    public @ResponseBody void putScopedKeyValue(HttpServletRequest request, HttpServletResponse response, @PathVariable String scope, @PathVariable String key, @RequestBody String valueJson) throws IOException {
+      String byUser = env.getRequiredProperty("scope." + scope + ".byUser");
+      boolean scopeUserBased = Boolean.parseBoolean(byUser);
+      logger.trace("scope " + scope + " byUser? " +byUser + " : " + scopeUserBased);
+
+      String username = request.getHeader(usernameAttribute);
+      
+      String prefix = scope;
+      boolean authorized = false;
+      
+      if(!scopeUserBased) {
+        //we are editing a global scoped thing, so we must be an admin
+        String adminGroup = env.getRequiredProperty("scope." + scope + ".admin.group");
+        String groupHeader = env.getRequiredProperty("groupHeaderAttribute");
+        
+        String header = request.getHeader(groupHeader);
+        
+        authorized = header !=null && header.contains(adminGroup);
+      } else {
+        //a user is PUTing on a scoped key that is per attribute
+        String prefixAttr = env.getRequiredProperty("scope." + scope + ".prefixAttribute");
+        String filterHeaderValue = request.getHeader(prefixAttr); 
+        authorized = filterHeaderValue != null;
+        prefix += ":" + filterHeaderValue;
+      }
```

Comment:
Since uw-spring-security is not open source, we can't use it for this project. We could possibly use vanilla spring-security. I think to be iterative on this we could:
+ Move the authorization to the service and keep the code we have now
+ Create an issue to refactor to use spring security for said authorization